Px.Utils/Validation/SyntaxValidation/SyntaxValidationUtilityMethods.cs

```diff
@@ -0,0 +1,404 @@
+    /// <param name="remainder">The remainder of the string after extraction.</param>
+    public readonly struct ExtractSectionResult(string[] parameters, string remainder)
+    {
+        /// <summary>
+        /// Gets the sections extracted from the string.
+        /// </summary>
+        public string[] Sections { get; } = parameters;
+
+        /// <summary>
+        /// Gets the remainder of the string after extraction.
+        /// </summary>
+        public string Remainder { get; } = remainder;
+    }
+
+    /// <summary>
+    /// Provides a collection of helper methods used during the syntax validation process. These methods include functionality for extracting sections from a string, checking the format of a string, and determining the type of a value from a string.
+    /// </summary>
+    public static class SyntaxValidationUtilityMethods
+    {
+        private const int DEFAULT_TIMEOUT = 1;
+
+        /// <summary>
+        /// Determines whether a string contains more than one section enclosed with given symbols.
+        /// </summary>
+        /// <param name="input">The input string to check</param>
+        /// <param name="startSymbol">Symbol that starts enclosement</param>
+        /// <param name="syntaxConf">Object that contains the symbols and tokens for structuring the file syntax. The syntax configuration is represented by a <see cref="PxFileSyntaxConf"/> object.</param>
+        /// <param name="endSymbol">Symbols that closes the enclosement</param>
+        /// <returns>Returns a boolean which is true if the input string contains more than one section</returns>
+        public static bool HasMoreThanOneSection(string input, char startSymbol, char endSymbol, PxFileSyntaxConf syntaxConf)
+        {
+            ExtractSectionResult searchResult = ExtractSectionFromString(input, startSymbol, syntaxConf, endSymbol);
+            return searchResult.Sections.Length > 1;
+        }
+
+        /// <summary>
+        /// Extracts a section from a string enclosed with given symbols.
+        /// </summary>
+        /// <param name="input">The input string to extract from</param>
+        /// <param name="startSymbol">Symbol that starts enclosement</param>
+        /// <param name="syntaxConf">Object that contains the symbols and tokens for structuring the file syntax. The syntax configuration is represented by a <see cref="PxFileSyntaxConf"/> object.</param>
+        /// <param name="optionalEndSymbol">Optional symbol that closes the enclosement. If none given, startSymbol is used for both starting and ending the enclosement</param>
+        /// <return>Returns an <see cref="ExtractSectionResult"/> object that contains the extracted sections and the string that remains after the operation</return>
+        public static ExtractSectionResult ExtractSectionFromString(string input, char startSymbol, PxFileSyntaxConf syntaxConf, char? optionalEndSymbol = null)
```

Comment:
Only string delimeter is used from conf, so that should be passed directly.
No need to name the parameter optional, the default value makes that clear.
I would change the input string to ref and only return the result array. This is to avoid wrapper types.